libraries/cyclestreets-view/src/main/java/net/cyclestreets/views/overlay/WaymarkOverlay.kt

```diff
@@ -0,0 +1,159 @@
+package net.cyclestreets.views.overlay
+
+import android.content.SharedPreferences
+import android.content.SharedPreferences.Editor
+import android.graphics.Canvas
+import android.graphics.Matrix
+import android.graphics.Paint
+import android.graphics.Point
+import android.graphics.drawable.BitmapDrawable
+import android.graphics.drawable.Drawable
+import android.support.v4.content.res.ResourcesCompat
+
+import net.cyclestreets.routing.Journey
+import net.cyclestreets.routing.Route
+import net.cyclestreets.routing.Waypoints
+import net.cyclestreets.view.R
+import net.cyclestreets.views.CycleMapView
+
+import org.osmdroid.api.IGeoPoint
+import org.osmdroid.api.IProjection
+import org.osmdroid.util.GeoPoint
+import org.osmdroid.views.MapView
+import org.osmdroid.views.overlay.Overlay
+import org.osmdroid.views.overlay.OverlayItem
+
+import java.util.ArrayList
+
+class WaymarkOverlay(private val mapView: CycleMapView) : Overlay(), PauseResumeListener, Route.Listener {
+
+    private val greenWisp: Drawable?
+    private val orangeWisp: Drawable?
+    private val redWisp: Drawable?
+    private val screenPos = Point()
+    private val bitmapTransform = Matrix()
+    private val bitmapPaint = Paint()
+
+    private val waymarkers: MutableList<OverlayItem>
+
+    init {
+        val res = mapView.context.resources
+        val makeWisp = { drawable: Int -> ResourcesCompat.getDrawable(res, drawable, null) }
+
+        greenWisp = makeWisp(R.drawable.green_wisp)
+        orangeWisp = makeWisp(R.drawable.orange_wisp)
+        redWisp = makeWisp(R.drawable.red_wisp)
+
+        waymarkers = ArrayList()
```

Comment:
I suggest inlining this initialisation to where `waymarkers` is defined

Reply:
Doh